src/react/components.js

```diff
@@ -218,16 +224,16 @@ export function createClassInstanceForFirstRenderOnly(
     }
     if (stateToUpdate instanceof ObjectValue) {
       let newState = new ObjectValue(realm, realm.intrinsics.ObjectPrototype);
+      newState.makeFinal();
```

Comment:
How do we make it final and then immediately do an `Object.assign` to it? Seems inconsistent. I thought finals are treated as immutable.

Should we do this *after* it's fully initialized?

Reply:
Yes, you beat me to this, I changed it in my latest commit.